src/codec.rs

```diff
@@ -0,0 +1,43 @@
+//! This module contains traits to have a unified API across codecs.
+//!
+//! There are two traits defined, [`Codec`] and [`Links`]. Those are separate traits as the `Links`
+//! trait is not generic over a certain type.
+
+use cid::Cid;
+
+use std::io::{BufRead, Write};
+
+/// Each IPLD codec implementation should implement this Codec trait. This way codecs can be more
+/// easily exchanged or combined.
+pub trait Codec<T>: Links {
+    /// The multicodec code of the IPLD codec.
+    const CODE: u64;
```

Comment:
I think we're going to want `NAME` eventually too, it's pretty helpful this trait is almost the same as the interfaces we landed on in JS, although we ended up splitting encode and decode operations into two parts and combining them into a single `BlockCodec`: https://github.com/multiformats/js-multiformats/blob/master/src/codecs/interface.ts

Reply:
I don't want the name. People should really use the constants as identifiers and not strings that might change.

Splitting encode and decode: I'd keep it simple for now, if the need of splitting ever occurs (which I currently doubt), we can do it then.